docs/experimental/auth_requirements_errors.rst

```diff
@@ -0,0 +1,136 @@
+Auth Requirements Errors
+========================
+
+Globus Auth Requirements Error is a response format that conveys to a client any
+modifications to a session (i.e., "boosting") that will be required
+to complete a particular request.
+
+The ``auth_requirements_error`` module provides a number of tools to make it easier to
+identify and handle these errors when they occur.
+
+GlobusAuthRequirementsError
+---------------------------
+
+The ``GlobusAuthRequirementsError`` class provides a model for working with Globus
+Auth Requirements Error responses. The shape of an instance closely
+matches that of the JSON response, such that in order to access a
+response's required_scopes one could use, e.g.,:
+
+.. code-block:: python
+
+    from globus_sdk.experimental import auth_requirements_error
+
+    error = auth_requirements_error.GlobusAuthRequirementsError(response)
+    error.authorization_parameters.required_scopes
+
+``GlobusAuthRequirementsError`` enforces types strictly when parsing a Globus
+Auth Requirements Error response dictionary, and will raise a ``ValueError`` if a
+supported field is supplied with a value of the wrong type.
+``GlobusAuthRequirementsError`` does not, however, attempt to mimic or itself enforce
+any logic specific to the Globus Auth service with regard to what represents a valid
+combination of fields (e.g., ``session_required_mfa`` requires either
+``session_required_identities`` or ``session_required_single_domain``
+in order to be properly handled).
```

Comment:
I think this is a great idea. I'll ruminate on whether there are enough additional errata that it warrants a full section. If not, I still think your point is very valid about breaking up the flow with details. In that case, perhaps we move to an end-of-section note so that it's not interrupting the narrative? Will follow up in a commit either way.